src/python_pachyderm/experimental/mixin/pfs.py

```diff
@@ -1254,6 +1258,93 @@ def path_exists(
 
         return True
 
+    def mount(self, mount_dir: str, repos: List[str] = []) -> None:
+        """Mounts Pachyderm repos locally.
+
+        Parameters
+        ----------
+        mount_dir : str
+            The directory to mount repos to. Make sure if this folder already
+            exists that it's empty (including hidden files).
+        repos : List[str], optional
+            The repos to mount. Each repo can only be mounted once, even if
+            multiple branches are passed. If empty, all repos are mounted.
+
+        Notes
+        -----
+        Mounting uses FUSE, which causes some known issues on macOS. For the
+        best experience, we recommend using mount on Linux. We do not support
+        mounting on macOS 1.11 and later.
+
+        Additionally, we recommend using mount in read-only access.
+
+        Examples
+        --------
+        >>> client.mount("dir_a", ["repo1", "repo2@staging"])
+        """
+        Path(mount_dir).mkdir(parents=True, exist_ok=True)
+
+        subprocess.run(
+            ["sudo", "pachctl", "unmount", mount_dir],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.STDOUT,
+        )
+
+        # Check for non-empty mount dir
+        mount_dir_contents = next(os.walk(mount_dir))
+        if mount_dir_contents[1] or mount_dir_contents[2]:
+            raise RuntimeError(
+                f"{mount_dir} must be empty to mount (including hidden files)"
+            )
+
+        # If 0 Pachyderm repos, no need to mount
+        if not list(self.list_repo()):
+            print("no repos in Pachyderm to mount")
+            return
+
+        cmd = ["pachctl", "mount", mount_dir]
+        for r in repos:
+            cmd.append("-r")
+            cmd.append(r)
+
+        subprocess.Popen(cmd, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+
+        # Ensure mount has finished
+        for _ in range(3):
+            time.sleep(0.25)
+            mounted_repos = next(os.walk(mount_dir))[1]
+
+            if mounted_repos:
+                return
+
+        self.unmount(mount_dir)
```

Comment:
why does mount() call unmount()?

Reply:
I noticed a small delay between `pachctl mount` and the actual files showing up in the local directory. Thus, I added a check in the `for` loop to make sure the files actually got mounted. In the rare state that the check fails all 3 times, it `pachctl unmount`s to undo the `mount`.

Reply:
got it - thanks! i missed the early exit.

Reply:
would this still work if the repo was empty?

Reply:
yep!